src/app/employee/employee-form/employee-form.component.ts

```diff
@@ -19,4 +28,18 @@ export class EmployeeFormComponent extends BaseForm {
     return new FormGroup({});
   }
 
+  ngOnInit() {
+    if (this.activatedRoute.snapshot.paramMap.has('id')) {
+      this.subs.push(
+        this.activatedRoute.paramMap.pipe(
+          switchMap((params: ParamMap) => this.employeesService.getById(Number.parseInt(params.get('id'))))
+        ).subscribe((employee: IEmployee) => this.employee$.next(employee))
+      )
+    }
+  }
+
+  ngOnDestroy() {
+    this.subs.forEach(s => s.unsubscribe())
```

Comment:
We've started using this simple new tool at DIV4 called [SubSink](https://github.com/wardbell/subsink/blob/ea8813e0850200afc06cde54f7fdc87ae5e98a26/README.md). It essentially just encapsulates all the unsubscription logic. Might not necessarily need to add to the TCP app since it's not as obvious how it works but you all may find it useful on other projects.

Reply:
I'm familiar with it.  I was feeling lazy about pulling in a package when the pattern is not too hard to implement oneself.

We could easily pull it in and use here.  A refactor to do when we have some free time.